File: js/fibonacci.js
const fibonacci = (num) => {
    let fibonacciSequence = [0, 1];

    if (num === 0 || num === 1) {
        return fibonacciSequence[num]
    } else {
        let prev = fibonacciSequence[0];
        let current = 0;

        for (let i = 1; i <= fibonacciSequence.length; i++) {
            if (fibonacciSequence.length != num + 1) {
                current = prev + fibonacciSequence[i];
                prev = fibonacciSequence[i]
                fibonacciSequence.push(current);
            }    
        }
        return fibonacciSequence.pop()
    }
}

module.exports = {fibonacci}

File: python/fibonacci.py
def fibonacci(n):
  if n == 0 or n == 1:
    return n
  
  arr = [0, 1]
  previous = arr[0]
  current = arr[1]

  for x in range(1, n + 1, 1):
    while len(arr) != (n + 1):
        next = previous + current
        previous = current
        current = next
        arr.append(current)
  return arr[-1]